WordPress/wordcamp.org
# Blocks: Create a Live Schedule block

Add a new block called "Live Schedule", which can be used on the Day of Event template (or any other page). Combined with #241, this will let organizers have total control over the content of the Day of Event template.

This also removes the need for a separate build script/process in `mu-plugins/theme-templates`, that will be removed in a following PR. It does remove the self-updating "Latest Posts" section, but organizers can use the Latest Posts block if they want, and this reduces the repeat API calls. If more organizers want this functionality we can see about hooking into the Latest Posts block.

Fixes #127 – You can customize the heading text and level in the block UI.
See #130 – Addresses a few issues from this issue

**Screenshots**

In wp-admin, it only displays a placeholder (for now). The headings are customized in the sidebar. 

![Screen Shot 2019-09-27 at 6 46 16 PM](https://user-images.githubusercontent.com/541093/65806363-2fb05380-e157-11e9-9cc1-b5e133f96ba2.png)

No sessions upcoming:

![Screen Shot 2019-09-27 at 6 51 33 PM](https://user-images.githubusercontent.com/541093/65806545-05ab6100-e158-11e9-8253-fdfcd22c0965.png)

List of sessions (faked "now" to be 10am):

![Screen Shot 2019-09-27 at 6 56 07 PM](https://user-images.githubusercontent.com/541093/65806628-7e122200-e158-11e9-99be-4e0805ffd110.png)

-----

Still to do (in following PRs)

- Fix issues with time
  - If we're after the start of the last event for a day, it says nothing's happening (we could check if there were other sessions on this day, and put up a "done for the day" message?)
  - If we're before the first event on the second day, it considers the last session of the prev day "on now".
- Remove `stripTagsAndEncodeText` and support diacritics.
- Audit the imports from the frontend script so we're not loading too much to the page
- Add a better, interactive preview in the editor
- (as mentioned) Remove the JS code from day of theme-templates.

------

**To test**

- Create a few sessions that are happening now (or add a property to `$data['live-schedule']` called `nowOverride` which should be a fake "now" in miliseconds).
- Add the Live Schedule block to a page.
- View it on the frontend, it should show you the current and upcoming sessions.
